flink-runtime/src/test/java/org/apache/flink/runtime/io/network/api/serialization/SpanningRecordSerializationTest.java

```diff
@@ -104,11 +120,236 @@ public void testHandleMixedLargeRecords() throws Exception {
 		testSerializationRoundTrip(originalRecords, segmentSize);
 	}
 
+	/**
+	 * Non-spanning, deserialization reads one byte too many and succeeds.
+	 */
+	@Test
+	public void testHandleDeserializingTooMuchNonSpanning1() throws Exception {
+		testHandleWrongDeserialization(
+			new StringValueDeserializingTooMuch("Test string"),
+			32 * 1024,
+			null);
+	}
+
+	/**
+	 * Non-spanning, serialization length is 16 (including headers), deserialization reads one byte
+	 * too many and succeeds.
+	 */
+	@Test
+	public void testHandleDeserializingTooMuchNonSpanning2() throws Exception {
+		testHandleWrongDeserialization(
+			new StringValueDeserializingTooMuch("Test string"),
+			17,
```

Comment:
yes, that would be really nice but is not available out of the box since `StringValue#getBinaryLength` only returns `-1` and additionally, we do add one more length header from the `SpanningRecordSerializer` - I could evaluate the actual value through serializing it (maybe only once, statically) though and keep using that or so...